ractor_actors/examples/tcp.rs

```diff
@@ -0,0 +1,205 @@
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree.
+
+extern crate ractor_actors;
+
+use chrono::{DateTime, Utc};
+use ractor::{Actor, ActorProcessingErr, ActorRef, DerivedActorRef};
+use ractor_actors::net::tcp::listener::*;
+use ractor_actors::net::tcp::session::*;
+use ractor_actors::net::tcp::*;
+use ractor_actors::watchdog;
+use ractor_actors::watchdog::TimeoutStrategy;
+use std::error::Error;
+use std::str::FromStr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::SystemTime;
+
+/// Controls if the watchdog is used
+static WATCHDOG: AtomicBool = AtomicBool::new(false);
+
+struct MyServer {}
+
+impl Actor for MyServer {
+    type Msg = ();
+    type State = ();
+    type Arguments = NetworkPort;
+
+    async fn pre_start(
+        &self,
+        myself: ActorRef<Self::Msg>,
+        port: Self::Arguments,
+    ) -> Result<Self::State, ActorProcessingErr> {
+        let _ = myself
+            .spawn_linked(
+                Some(format!("listener-{}", port)),
+                Listener::new(
+                    port,
+                    IncomingEncryptionMode::Raw,
+                    move |stream| async move {
+                        tracing::info!("New connection: {}", stream.peer_addr());
+                        Actor::spawn(
+                            Some(format!("MySession-{}", stream.peer_addr().port())),
+                            MySession {
+                                info: stream.info(),
+                            },
+                            stream,
+                        )
+                        .await
+                        .map_err(|e| ActorProcessingErr::from(e))?;
+                        Ok(())
+                    },
+                ),
+                (),
+            )
+            .await?;
+
+        Ok(())
+    }
+}
+
+struct MySession {
+    info: NetworkStreamInfo,
+}
+
+enum MySessionMsg {
+    FrameAvailable(Frame),
+}
+
+impl From<FrameAvailable> for MySessionMsg {
+    fn from(FrameAvailable(frame): FrameAvailable) -> Self {
+        Self::FrameAvailable(frame)
+    }
+}
+
+impl TryFrom<MySessionMsg> for FrameAvailable {
+    type Error = ();
+
+    fn try_from(_: MySessionMsg) -> Result<Self, Self::Error> {
+        Err(())
+    }
+}
+
+struct MySessionState {
+    session: DerivedActorRef<SendFrame>,
+}
+
+impl Actor for MySession {
+    type Msg = MySessionMsg;
+    type State = MySessionState;
+    type Arguments = NetworkStream;
+
+    async fn pre_start(
+        &self,
+        myself: ActorRef<Self::Msg>,
+        stream: Self::Arguments,
+    ) -> Result<Self::State, ActorProcessingErr> {
+        tracing::info!("New session: {}", stream.peer_addr());
+
+        let session = Session::spawn_linked(myself.get_derived(), stream, myself.get_cell())
+            .await
+            .map_err(ActorProcessingErr::from)?;
+
+        if WATCHDOG.load(Ordering::SeqCst) {
```

Comment:
this really should just be an arg to this actor, there's no reason it needs to be a static atomic bool

Reply:
Yep, fixed in 9e2c53e2.